### ISSUE TITLE
how to aggregate online count?

### ISSUE THREAD

**Issue by funwun** [pipelinedb/pipelinedb#1815]
I have device online offline events, and insert into a stream like this:
```
create stream online_stream (
    id varchar(64),
    ts timestamptz,
    status smallint -- 0: offline, 1: online
);
```
I use this to get the status of all devices:
```
create continuous view online_view as
    select
    keyed_max(ts, status) as status,
    max(ts) as ts,
    id
    from mqtt_online_stream group by id;
```

But how to get the online device count for each 1 minute?

Thanks

**Comment by usmanm** [comment]
You'll probably have to create a different view since you're now aggregating over `id` and a time window.

```
CREATE CONTINUOUS VIEW online_counts AS
  SELECT
    id,
    minute(ts),
    sum(status) AS count -- this works because 0 is the status for offline, otherwise you could also use a CASE statement here
  FROM online_stream
  GROUP BY id, minute(ts); 
```

**Comment by funwun** [comment]
Can I have the min and max count for each minutes?

**Comment by usmanm** [comment]
Not sure if I follow. Do you mean you want to keep track of what's the least number of times you've seen a device online per minute vs what's the most number of times you've seen it per minute?

**Comment by funwun** [comment]
Yes, and other problem, my system will send the online message more than one times, same as offline message, so I create the online_view to keep the status of each device.

I can select count() of the online_view where status = 1.

**Comment by usmanm** [comment]
Could you describe the system you're trying to build at a higher level? That might be helpful for me to understand what you're trying to accomplish with your queries.

**Comment by funwun** [comment]
When the device online, the system will insert the device is and status =1 into the online_stream, but I can't guarantee the insert one time, because after system crash, it will send online again.

**Comment by funwun** [comment]
@usmanm do you have any suggestion?
